### PR TITLE
Ft 16 losing tokens when plugin restarts

### PR DESCRIPTION
**Pull request by swordEdge**

### CODE REVIEW

**Review by LiamMartens — APPROVED** [review]

File: src/app/components/App.test.tsx [hunk 1]
```diff
@@ -36,41 +38,13 @@ describe('App', () => {
...
       }),
     );
-    const WelcomeText = getByText('Welcome to Figma Tokens.');
 
-    expect(WelcomeText).toBeInTheDocument();
-  });
+    const cancelButtons = getAllByText('Cancel');
+    fireEvent.click(cancelButtons[1]);
 
-  it('skip to token screen when there is a token', () => {
```
Comment (six7) [inline src/app/components/App.test.tsx:44]: Can we make sure to still keep this test that the loading screen advances by itself when tokens are found? You removed this and added a cancel click but both cases need to work

File: src/app/components/Initiator.tsx [hunk 2]
```diff
@@ -22,14 +23,23 @@ export function Initiator() {
...
   const licenseKey = useSelector(licenseKeySelector);
   const checkedLocalStorage = useSelector(checkedLocalStorageForKeySelector);
-
-  const onInitiate = () => {
-    postToFigma({ type: MessageToPluginTypes.INITIATE });
-  };
   const userId = useSelector(userIdSelector);
 
+  const askUserIfPull: (() => Promise<any>) = React.useCallback(async () => {
+    const shouldPull = await confirm({
```
Comment (six7) [inline src/app/components/Initiator.tsx:33]: Its not just github users can pull from, its depending on what provider they have. Jsonbin, github, gitlab, azure devops

File: src/app/components/Initiator.tsx [hunk 3]
```diff
@@ -100,47 +125,44 @@ export function Initiator() {
...
+                  const remoteData = await pullTokens({ context: credentials, featureFlags: receivedFlags, usedTokenSet });
+                  const existTokens = Object.values(remoteData?.tokens ?? {}).some((value) => value.length > 0);
+  
+                  if (existTokens) { dispatch.uiState.setActiveTab(Tabs.TOKENS); } else { dispatch.uiState.setActiveTab(Tabs.START); }
+                } else {
+                  dispatch.uiState.setActiveTab(Tabs.TOKENS);
+                }
+              } else {
                 dispatch.uiState.setActiveTab(Tabs.START);
-                notifyToUI('Failed to fetch tokens, check your credentials', { error: true });
```
Comment (six7) [inline src/app/components/Initiator.tsx:140]: Did you remove that and move it somewhere else? Cant find it anymore. This is to tell the user that we were unable to fetch

File: src/app/store/updateSources.tsx [hunk 4]
```diff
@@ -31,6 +31,7 @@ type UpdateTokensOnSourcesPayload = {
   storageType: StorageType;
   lastUpdatedAt: string;
   api: ContextObject;
+  checkForChanges: string
```
Comment (six7) [inline src/app/store/updateSources.tsx:34]: This should give an eslint error and be autofixable

File: src/types/tokens/TokensStore.ts [hunk 5]
```diff
@@ -6,8 +6,9 @@ export type TokenStore = {
   version: string;
   updatedAt: string;
   // @README these could be different themes or sets of tokens
-  values: Record<string, AnyTokenList>;
+  values: Record<string, AnyTokenList>
```
Comment (six7) [inline src/types/tokens/TokensStore.ts:9]: Same, should give eslint error

**Review by six7 — CHANGES_REQUESTED** [review]

**Comment by six7** [comment]
functionally this works fine! if you could merge `next` back in, we had some changes with release-99 that touched the same files i believe. once merged next and confirmed working we can merge it

**Comment by six7** [comment]
@swordEdge once you fix the failing test we can merge

**Pull request merged**